src/visualization/recipes_makie.jl

```diff
@@ -102,8 +102,16 @@ function iplot(pd::PlotData2DTriangulated;
   flat_wire_points = Makie.@lift get_flat_points($wire_points, $z_offset)
   wire_mesh_flat = Makie.lines!(ax, flat_wire_points, color=:black)
 
+  # create a small variation in the extrema to avoid the Makie `range_step` cannot be zero error.
+  # see https://github.com/MakieOrg/Makie.jl/issues/931 for more details.
+  # the colorbar range is perturbed by 1e-5 * the magnitude of the solution.
+  function scaled_extrema(x)
+    ex = extrema(x) 
+    return ex .+ 1e-5 .* maximum(abs.(ex)) .* (-1, 1)
```

Comment:
I do not like the fact that we "smear" out the colorbar limits for every plot. Right now, the color bar limits accurately indicate the actual limits of the plotted quantity. If we, for example, plot density in a range [1e-6, 1.0], this would now let the colorbar start in the negative, even if the solution is not.

Could we maybe instead use an approach where we only "fix" the limits in the constant case, i.e., where the extrema are approximately identical?

Reply:
Sounds reasonable as well :+1:

Reply:
Should be fixed in https://github.com/trixi-framework/Trixi.jl/pull/1259/commits/f404c0bc863e09dc7dc76f43c1091f70e927cd2a